src/data/polynomial/degree/basic.lean

```diff
@@ -202,6 +202,16 @@ begin
     finset.sum_mul_boole, finset_sum_coeff, ite_le_nat_degree_coeff],
 end
 
+lemma as_sum'(p : polynomial R) :
+  p = ∑ i in p.support, C (p.coeff i) * X^i :=
```

Comment:
Why do you want this instead of `as_sum`?

Reply:
Because if we use `support`, then all the coefficient is non-zero, hence division is less painful.

Reply:
But you are right that one can always use `as_sum`, if this is not worth having, I will just delete this lemma.

Reply:
Also I used it in a lemma to prove Liouville theorem, see https://github.com/leanprover-community/mathlib/pull/4301

Reply:
What do you think of calling this `as_sum_support` to make is a bit more clear? I agree that by symmetry the other one ought to be called `as_sum_range`.